srplasticity/srp_extrafuncs.py

```diff
@@ -390,91 +498,140 @@ def plot_srp(model, target_dict, stimulus_dict, protocols=None):
+
     axis.spines['top'].set_visible(False)
     axis.spines['right'].set_visible(False)
     axis.set_ylabel("Predicted EPSC", labelpad=1, size=8)
     axis.set_xticks(range(1, len(xax) + 1))
     axis.set_xlabel("spike nr.", labelpad=1)
-    fig.set_dpi(1200)
-    fig.tight_layout()
-    # plt.savefig(f"estimates_plot.svg", transparent=True)
+    fig.legend(frameon=False)
 
 
 def plot_spike_train(spiketrain):
+    """
+    Plot a spike train
+
+    :param spiketrain: A binary stimulation vector from a
+                        vector with ISI intervals to be plotted
+    :type spiketrain: Numpy array
+    """
+
     fig, axis = plt.subplots()
     axis.plot(spiketrain, lw=0.7, color='black')
-    axis.set_ylim(-0.005, 0.005)
+    axis.set_ylim(-1e-5, 5e-6)
     axis.axis("off")
 
-    fig.set_dpi(1200)
-    fig.tight_layout()
-    # plt.savefig(f"spike_train_plot.svg", transparent=True)
 
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #
-# HELPER FUNCTIONS FOR FITTING PROCEDURE
+# HELPER FUNCTIONS
 #
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
 
-def _total_loss_det(target_vals, mean_predicted):
-
-    """
-    Stand in Mean Squared error for training loss in first phase
-    :param target_vals: (np.array) set of amplitudes
-    :param mean_predicted: (np.array) set of means
-    """
-
-    loss = []
-
-    for key in target_vals.keys():
-        for i in range(0, len(target_vals[key])):
-            run_arr = target_vals[key][i]  # get amplitudes from a single run
-            run_err = []
-
-            if not np.isscalar(run_arr):
-                for j in range(0, len(run_arr)):
-                    run_err.append(math.pow((run_arr[j] - mean_predicted[key][j]), 2))
-                loss.append(run_err)
-    
-    #this section is unneccessary and confusing: numpy already flattens the array
-    loss_2 = []
-    for i in loss:
-        for j in i:
-            loss_2.append(j)
-
-    total_mse_loss = np.nanmean(loss_2)
-
-    return total_mse_loss
-
 def get_poisson_ISIs(nspikes, rate):
     """
-    poisson ISIs
+    Poisson ISIs
+
     :param nspikes: number of spikes
+    :type nspikes: int
     :param rate: firing rate in Hz
+    :type rate: int
     """
 
     meanISI = int(1000 / rate)
     isis = np.random.exponential(scale=meanISI, size=nspikes).round(1)
     isis[isis < 2] = 2  # minimum 2ms refractory period
     return isis
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+#
+# PREPROCESSING
+#
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+
+def norm_responses(target_dict):
+    """
+    Normalizes the responses in the `target_dict` by averaging the first responses values,
+    then dividing all responses by this average value.
+
+    :param target_dict: A dictionary where keys are protocol names and values are NumPy arrays of the responses
```

Comment:
The dict format needs to be fully defined, especially since this is likely a dict of dicts of arrays or something similar.

Reply:
Done